shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala

```diff
@@ -324,7 +324,7 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
       conf: SQLConf): RDD[CachedBatch] = {
 
     val rapidsConf = new RapidsConf(conf)
-    val bytesAllowedPerBatch = rapidsConf.bytesAllowedPerBatch
+    val bytesAllowedPerBatch = rapidsConf.gpuTargetBatchSizeBytes - APPROX_PAR_META_DATA
```

Comment:
Do we need to max this with `1` or something to prevent it from going negative?  This comment applies to the other places where this computation is done below.

Reply:
Good point. Let me add a min batch size in the code